Java Client breaks on parsing GetAliasesResponse when one alias does not exist

### Elasticsearch Version

8.18.1

### Installed Plugins

_No response_

### Java Version

_bundled_

### OS Version

docker image: elasticsearch/elasticsearch:8.18.1

### Problem Description

When you use the `_alias` endpoint to resolve multiple aliases at once, the Java client breaks on parsing the response if one of them does not exist.

**Expected behaviour:**

GetAliasesResponse containing alias/index relations that do exist

**Actual behaviour:**

A TransportException is thrown and the existing aliases go unnoticed.





### Steps to Reproduce


```
PUT /example-index
PUT /example-index/_alias/test

GET /_alias/test,test2
```

The response will look like this:

```
{
  "error": "alias [test2] missing",
  "status": 404,
  "example-index": {
    "aliases": {
      "test": {}
    }
  }
}
```


So this response tells us that `test2` does not exist, and it also tells us that `example-index` has the alias `test`. However, the Java client breaks on this because the response has a HTTP 404 response, and error but also the regular payload. This leaves us with a `TransportException` rather than an `ElasticsearchException`, or, preferably, a `GetAliasesResponse`.

The cat/_aliases endpoint gives usable response with 

```
GET /_cat/aliases/test,test2?format=json&h=index,alias
```

but the docs clearly state that that is not intended for software consumption.

### Logs (if relevant)

```
co.elastic.clients.transport.TransportException: node: http://localhost:9201/, status: 404, [es/indices.get_alias] Failed to decode error response, check exception cause for additional details
	at co.elastic.clients.transport.ElasticsearchTransportBase.getApiResponse(ElasticsearchTransportBase.java:369)
	at co.elastic.clients.transport.ElasticsearchTransportBase.lambda$performRequestAsync$0(ElasticsearchTransportBase.java:205)
	at java.base/java.util.concurrent.CompletableFuture.uniHandle(CompletableFuture.java:934)
	at java.base/java.util.concurrent.CompletableFuture$UniHandle.tryFire(CompletableFuture.java:911)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
	at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2179)
	at co.elastic.clients.transport.rest_client.RestClientHttpClient$1.onSuccess(RestClientHttpClient.java:116)
	at org.elasticsearch.client.RestClient$FailureTrackingResponseListener.onSuccess(RestClient.java:680)
	at org.elasticsearch.client.RestClient$1.completed(RestClient.java:403)
	at org.elasticsearch.client.RestClient$1.completed(RestClient.java:397)
	at org.apache.http.concurrent.BasicFuture.completed(BasicFuture.java:122)
	at org.apache.http.impl.nio.client.DefaultClientExchangeHandlerImpl.responseCompleted(DefaultClientExchangeHandlerImpl.java:182)
	at org.apache.http.nio.protocol.HttpAsyncRequestExecutor.processResponse(HttpAsyncRequestExecutor.java:448)
	at org.apache.http.nio.protocol.HttpAsyncRequestExecutor.inputReady(HttpAsyncRequestExecutor.java:338)
	at org.apache.http.impl.nio.DefaultNHttpClientConnection.consumeInput(DefaultNHttpClientConnection.java:265)
	at org.apache.http.impl.nio.client.InternalIODispatch.onInputReady(InternalIODispatch.java:87)
	at org.apache.http.impl.nio.client.InternalIODispatch.onInputReady(InternalIODispatch.java:40)
	at org.apache.http.impl.nio.reactor.AbstractIODispatch.inputReady(AbstractIODispatch.java:114)
	at org.apache.http.impl.nio.reactor.BaseIOReactor.readable(BaseIOReactor.java:162)
	at org.apache.http.impl.nio.reactor.AbstractIOReactor.processEvent(AbstractIOReactor.java:337)
	at org.apache.http.impl.nio.reactor.AbstractIOReactor.processEvents(AbstractIOReactor.java:315)
	at org.apache.http.impl.nio.reactor.AbstractIOReactor.execute(AbstractIOReactor.java:276)
	at org.apache.http.impl.nio.reactor.BaseIOReactor.execute(BaseIOReactor.java:104)
	at org.apache.http.impl.nio.reactor.AbstractMultiworkerIOReactor$Worker.run(AbstractMultiworkerIOReactor.java:591)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: co.elastic.clients.json.UnexpectedJsonEventException: Unexpected JSON event 'START_OBJECT' instead of 'KEY_NAME'
	at co.elastic.clients.json.JsonpUtils.expectEvent(JsonpUtils.java:106)
	at co.elastic.clients.transport.endpoints.EndpointBase$1.deserialize(EndpointBase.java:150)
	at co.elastic.clients.transport.endpoints.EndpointBase$1.deserialize(EndpointBase.java:144)
	at co.elastic.clients.json.JsonpDeserializer.deserialize(JsonpDeserializer.java:77)
	at co.elastic.clients.transport.ElasticsearchTransportBase.getApiResponse(ElasticsearchTransportBase.java:355)
	... 24 more
```

--- a/java-client/src/main/java/co/elastic/clients/transport/endpoints/EndpointBase.java
+++ b/java-client/src/main/java/co/elastic/clients/transport/endpoints/EndpointBase.java
@@ -21,6 +21,7 @@ package co.elastic.clients.transport.endpoints;
 
 import co.elastic.clients.elasticsearch._types.ErrorCause;
 import co.elastic.clients.elasticsearch._types.ErrorResponse;
+import co.elastic.clients.json.JsonData;
 import co.elastic.clients.json.JsonpDeserializer;
 import co.elastic.clients.json.JsonpDeserializerBase;
 import co.elastic.clients.json.JsonpMapper;
@@ -145,6 +146,8 @@ public class EndpointBase<RequestT, ResponseT> implements Endpoint<RequestT, Res
             @Override
             public ErrorResponse deserialize(JsonParser parser, JsonpMapper mapper, JsonParser.Event event) {
                 ErrorResponse.Builder builder = new ErrorResponse.Builder();
+                ErrorCause.Builder errorCauseBuilder = new ErrorCause.Builder();
+                ErrorCause errorCause = null;
                 builder.status(statusCode);
                 while ((event = parser.next()) != JsonParser.Event.END_OBJECT) {
                     JsonpUtils.expectEvent(parser, JsonParser.Event.KEY_NAME, event);
@@ -152,11 +155,11 @@ public class EndpointBase<RequestT, ResponseT> implements Endpoint<RequestT, Res
                         case "error":
                             switch (event = parser.next()) {
                                 case VALUE_STRING:
-                                    builder.error(e -> e.reason(parser.getString()).type("http_status_" + statusCode));
+                                    errorCauseBuilder.reason(parser.getString()).type("http_status_" + statusCode);
                                     break;
                                 default:
                                     JsonpUtils.expectEvent(parser, JsonParser.Event.START_OBJECT, event);
-                                    builder.error(ErrorCause._DESERIALIZER.deserialize(parser, mapper, event));
+                                    errorCause = ErrorCause._DESERIALIZER.deserialize(parser, mapper, event);
                                     break;
                             }
                             break;
@@ -164,9 +167,22 @@ public class EndpointBase<RequestT, ResponseT> implements Endpoint<RequestT, Res
                             JsonpUtils.expectNextEvent(parser, JsonParser.Event.VALUE_NUMBER);
                             builder.status(parser.getInt());
                             break;
+                        // could be additional information, deserializing it into error cause's metadata
+                        // for example 404 on GetAliasRequest
+                        default:
+                            String key = parser.getString();
+                            event = parser.next();
+                            JsonpUtils.expectEvent(parser, JsonParser.Event.START_OBJECT, event);
+                            errorCauseBuilder.metadata(key, JsonData._DESERIALIZER.deserialize(parser, mapper));
+                            break;
                     }
                 }
-
+                if (errorCause != null) {
+                    builder.error(errorCause);
+                }
+                else {
+                    builder.error(errorCauseBuilder.build());
+                }
                 return builder.build();
             }
         };

